File: Lab/cse655/C++/Decl.cpp
#include "Decl.h"

extern Tokenizer t;

void Decl::parseDecl()
{
  curToken=t.getToken();
  if(curToken != 4)
  {
    cout<<"Error - expecting int"<<endl; 
    exit(1);
  }
  t.skipToken();
  idl.parseIDListForDS();

  curToken=t.getToken();
  if(curToken != 12)
  {
    cout<<"Error - expecting ;"<<endl; 
    exit(1);
  }
  t.skipToken();
}

void Decl::printDecl()
{
  idl.printIDList();
}

void Decl::execDecl()
{
}

File: Lab/Lab2/Aniket/interpreter/Decl.java
import java.io.*;

//class for non-terminal <decl>
class Decl {
    private IdList idList;
    public Decl() {
        idList = null;
    }
    public void parseDecl() throws FileNotFoundException, IOException {
        Tokenizer t = Tokenizer.instance();
        
        //check and skip terminal "int"
        if(4 != t.getToken()) {
            System.out.println("Parse Error : expected : int");
            System.exit(0);
        }
        t.skipToken();

        idList = new IdList();
        idList.parseIdListForDS();

        //check and skip ;
        if(12 != t.getToken()) {
            System.out.println("Parse Error : expected : ;");
            System.exit(0);
        }
        t.skipToken();
    }
    
    public void printDecl() throws FileNotFoundException, IOException {
        System.out.print(String.format("%1$#4sint ", " "));
        idList.printIdList();
        System.out.println(";");
    }
}
